File: my_mindhelp_app/lib/models/counseling_center.dart
class CounselingCenter {
  final String id;
  final String name;
  final String address;
  final String phone;
  final bool onlineCounseling;
  final double? latitude;
  final double? longitude;
  final String createdAt;
  final String updatedAt;

  CounselingCenter({
    required this.id,
    required this.name,
    required this.address,
    required this.phone,
    required this.onlineCounseling,
    this.latitude,
    this.longitude,
    required this.createdAt,
    required this.updatedAt,
  });

  factory CounselingCenter.fromJson(Map<String, dynamic> json) {
    return CounselingCenter(
      id: json['id'] as String,
      name: json['name'] as String,
      address: json['address'] as String,
      phone: json['phone'] as String,
      onlineCounseling: json['online_counseling'] as bool,
      latitude: (json['latitude'] == null)
          ? null
          : (json['latitude'] is num
              ? (json['latitude'] as num).toDouble()
              : double.tryParse(json['latitude'].toString())),
      longitude: (json['longitude'] == null)
          ? null
          : (json['longitude'] is num
              ? (json['longitude'] as num).toDouble()
              : double.tryParse(json['longitude'].toString())),
      createdAt: json['created_at'] as String,
      updatedAt: json['updated_at'] as String,
    );
  }
}

File: backend/internal/models/counseling_center.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

// CounselingCenter 諮商所模型
type CounselingCenter struct {
	ID          uuid.UUID `gorm:"type:uuid;primaryKey" json:"id"`
	Name        string    `gorm:"size:255;not null" json:"name"`
	Description string    `gorm:"type:text" json:"description"`
	Address     string    `gorm:"type:text" json:"address"`
	Phone       string    `gorm:"size:50" json:"phone"`
	Email       string    `gorm:"size:255" json:"email"`
	Website     string    `gorm:"size:500" json:"website"`
	Latitude    float64   `json:"latitude"`
	Longitude   float64   `json:"longitude"`
	Services    string    `gorm:"type:text" json:"services"` // JSON 格式
	Hours       string    `gorm:"type:jsonb" json:"hours"` // JSON 格式
	PriceRange  string    `gorm:"size:100" json:"price_range"`
	Rating      float64   `gorm:"default:0" json:"rating"`
	ReviewCount int       `gorm:"default:0" json:"review_count"`
	IsVerified  bool      `gorm:"default:false" json:"is_verified"`
	IsActive    bool      `gorm:"default:true;index" json:"is_active"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	// 關聯
	Reviews []Review `gorm:"foreignKey:ResourceID;references:ID" json:"reviews,omitempty"`
}
